var/spack/repos/builtin/packages/tcl/package.py

```diff
@@ -49,12 +49,19 @@ def url_for_version(self, version):
         base_url = 'http://prdownloads.sourceforge.net/tcl'
         return '{0}/tcl{1}-src.tar.gz'.format(base_url, version)
 
-    def setup_environment(self, spack_env, run_env):
+    def setup_dependent_environment(self, spack_env, run_env, dependent_spec):
         # When using Tkinter from within spack provided python+tk, python
         # will not be able to find Tcl/Tk unless TCL_LIBRARY is set.
         run_env.set('TCL_LIBRARY', join_path(self.prefix.lib, 'tcl{0}'.format(
             self.spec.version.up_to(2))))
 
+    def install(self, spec, prefix):
+        with working_dir(self.build_directory):
+            make('install')
+
+            # Some applications like Expect require private Tcl headers.
+            make('install-private-headers')
```

Comment:
This is how Homebrew does it:
https://github.com/Homebrew/homebrew-dupes/blob/master/tcl-tk.rb#L78